agenthub/codeact_agent/codeact_agent.py

```diff
@@ -123,20 +123,42 @@ def action_to_str(self, action: Action) -> str:
             return action.content
         return ''
 
-    def get_action_message(self, action: Action) -> dict[str, str] | None:
+    def get_action_message(
+        self, action: Action
+    ) -> dict[str, str | MessageContent] | None:
         if (
             isinstance(action, AgentDelegateAction)
             or isinstance(action, CmdRunAction)
             or isinstance(action, IPythonRunCellAction)
-            or isinstance(action, MessageAction)
+            or (isinstance(action, MessageAction) and not action.images_base64)
```

Comment:
I'm thinking if we could organize the `MessageContent` a little bit better, e.g., have attributes like `.contains_img` that allows easy check like this.

Reply:
Makes sense.

Reply:
Specifically, i'm think if we build a `Message` pydantic class like OpenAI's API: https://github.com/openai/openai-python/blob/0de7419c962805bbe9db18d99524b66673f7f8e9/src/openai/types/chat/chat_completion_message.py#L25-L30

So that the `image_url` is supported as content?
https://github.com/openai/openai-python/blob/0de7419c962805bbe9db18d99524b66673f7f8e9/src/openai/types/chat/chat_completion_content_part_image_param.py#L10-L26

Though we should use litellm as a source of truth for the schema: https://litellm.vercel.app/docs/completion/vision but i think it should be identical to OpenAI's